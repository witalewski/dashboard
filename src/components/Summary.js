import React from 'react';
import styled from '@emotion/styled';

import { format } from 'date-fns';
import { HorizontalBar } from './charts/HorizontalBar';

const StyledSummary = styled.div`
  width: 100%;
  height: 100%;

  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;

  .main-label {
    font-weight: bold;
  }

  .main-sum {
    font-weight: bold;
    font-size: 36px;
  }

  .container--stretch {
    align-self: stretch;
  }

  .comparison {
    display: flex;
    justify-content: space-between;
  }

  .comparison__sum {
    font-weight: bold;
    font-size: 24px;
  }

  .percent-of-average-label {
    text-align: center;
    padding-bottom: 24px;
  }
`;

export const Summary = ({ data }) => (
  <StyledSummary>
    <div>
      <div className="main-label">In {format(new Date(), 'MMMM')} you did</div>
      <div className="main-sum">${data.revenueThisMonth}</div>
    </div>

    <div className="container--stretch">
      <div className="comparison">
        <div className="comparison__label">Last month you did:</div>
        <div className="comparison__sum">${data.revenueLastMonth}</div>
      </div>
      <div className="comparison">
        <div className="comparison__label">Average monthly revenue:</div>
        <div className="comparison__sum">${data.revenueAverage}</div>
      </div>
    </div>

    <div className="container--stretch">
      <HorizontalBar percent={data.revenueRate * 100} />
      <div className="percent-of-average-label">
        of your monthly average revenue
      </div>
    </div>
  </StyledSummary>
);
