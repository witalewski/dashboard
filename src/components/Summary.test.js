import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import mockChartData from '../fixtures/mockChartData.json';
import { Summary } from './Summary';

describe('Summary', () => {
  it('matches snapshot with mock data', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Summary data={mockChartData.summaryChart.data} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
