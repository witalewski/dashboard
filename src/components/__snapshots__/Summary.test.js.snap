// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary matches snapshot with mock data 1`] = `
<ForwardRef(render)>
  <div>
    <div
      className="main-label"
    >
      In 
      December
       you did
    </div>
    <div
      className="main-sum"
    >
      $
      100.64
    </div>
  </div>
  <div
    className="container--stretch"
  >
    <div
      className="comparison"
    >
      <div
        className="comparison__label"
      >
        Last month you did:
      </div>
      <div
        className="comparison__sum"
      >
        $
        199.09
      </div>
    </div>
    <div
      className="comparison"
    >
      <div
        className="comparison__label"
      >
        Average monthly revenue:
      </div>
      <div
        className="comparison__sum"
      >
        $
        131.11
      </div>
    </div>
  </div>
  <div
    className="container--stretch"
  >
    <HorizontalBar
      percent={38}
    />
    <div
      className="percent-of-average-label"
    >
      of your monthly average revenue
    </div>
  </div>
</ForwardRef(render)>
`;
